.github/workflows/update-dependency.yaml

```diff
@@ -0,0 +1,48 @@
+---
+name: 'Update Dependencies'
+on:
+  workflow_dispatch:
+  schedule:
+    # every Monday
+    - cron: '0 0 * * 1'
+env:
+  GOPROXY: https://proxy.golang.org
+permissions:
+  contents: read
+jobs:
+  update-deps:
+    if: ${{ github.repository == 'awslabs/amazon-eks-ami' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/setup-go@4.0.0
+        with:
+          go-version: '1.21.6'
+      - uses: actions/checkout@v4
+      - name: Update Nodeadm Dependencies
+        id: update_deps
+        run: |
+          cd nodeadm
+          make update-deps
+          echo 'changes<<EOF' >> $GITHUB_OUTPUT
+          git status --porcelain >> $GITHUB_OUTPUT
+          echo 'EOF' >> $GITHUB_OUTPUT
+      - name: Create PR
+        if: ${{ steps.update_deps.outputs.changes != '' }}
+        uses: peter-evans/create-pull-request@v5
+        with:
+          title: 'Update dependencies'
+          commit-message: Update dependencies
+          committer: github-actions <actions@github.com>
+          author: github-actions <actions@github.com>
+          branch: dependencies/update
+          branch-suffix: timestamp
+          base: master
+          delete-branch: true
+          labels: |
+            ok-to-test
+            release-note-none
```

Comment:
these labels aren't valid for our repo, we need to add `changelog/exclude` though.